### PR TITLE
Drag-and-drop improvements for projects and issue pins

### PR DESCRIPTION
**Pull request by silverwind**
1. Add "grabbing" cursor while dragging items:

2. Make project board only drag via their header, not via their whole body.

3. Fix some cursor problems in projects
4. Move shared options into `createSortable`.

### CODE REVIEW

**Comment by silverwind** [comment]
need to check "no permission" state

**Comment by silverwind** [comment]
Signed out state looks good already, no "grab" cursor anywhere, so this is ready for review.

File: web_src/js/modules/sortable.js [hunk 1]
```diff
@@ -1,4 +1,19 @@
...
   const {Sortable} = await import(/* webpackChunkName: "sortablejs" */'sortablejs');
-  return new Sortable(...args);
+
+  return new Sortable(el, {
+    animation: 150,
+    ghostClass: 'card-ghost',
+    onChoose: (e) => {
+      const handle = opts?.handle ? e.item.querySelector(opts.handle) : e.item;
+      handle.classList.add('tw-cursor-grabbing');
+      opts.onStart?.(e);
```
Comment (silverwind) [inline web_src/js/modules/sortable.js]:
```suggestion
      opts.onChoose?.(e);
```

**Comment by silverwind** [comment]
This will conflict with https://github.com/go-gitea/gitea/pull/29874 so should land that before this.

File: templates/projects/view.tmpl [hunk 2]
```diff
@@ -67,7 +67,7 @@
 	<div class="board {{if .CanWriteProjects}}sortable{{end}}">
 		{{range .Columns}}
 			<div class="ui segment project-column" style="background: {{.Color}} !important;" data-id="{{.ID}}" data-sorting="{{.Sorting}}" data-url="{{$.Link}}/{{.ID}}">
-				<div class="project-column-header">
+				<div class="project-column-header{{if and $canWriteProject (ne .ID 0)}}{{/* ID 0 is default column which cannot be moved */}} tw-cursor-grab{{end}}">
```
Comment (delvh) [inline templates/projects/view.tmpl]: Condition won't be needed with https://github.com/go-gitea/gitea/pull/29874 anymore.

**Review by delvh — COMMENTED** [review]
How will this PR interact with https://github.com/go-gitea/gitea/pull/29874?

File: templates/projects/view.tmpl (hunk 2 above)
Comment (silverwind) [reply templates/projects/view.tmpl]: Yes, that's the only "interaction" it has and it would cause a merge conflict anyways.

**Comment by silverwind** [comment]
> How will this PR interact with #29874?

No interactions besides the conditional in https://github.com/go-gitea/gitea/pull/29875#discussion_r1536922229.

File: templates/projects/view.tmpl (hunk 2 above)
Comment (silverwind) [reply templates/projects/view.tmpl]: Conflict fixed.

**Comment by silverwind** [comment]
Ready

File: web_src/js/modules/sortable.js [hunk 3]
```diff
@@ -1,4 +1,19 @@
-export async function createSortable(...args) {
+export async function createSortable(el, opts = {}) {
   const {Sortable} = await import(/* webpackChunkName: "sortablejs" */'sortablejs');
-  return new Sortable(...args);
+
+  return new Sortable(el, {
+    animation: 150,
+    ghostClass: 'card-ghost',
+    onChoose: (e) => {
+      const handle = opts?.handle ? e.item.querySelector(opts.handle) : e.item;
```
Comment (silverwind) [inline web_src/js/modules/sortable.js]:
```suggestion
      const handle = opts.handle ? e.item.querySelector(opts.handle) : e.item;
```

File: web_src/js/modules/sortable.js [hunk 4]
```diff
@@ -1,4 +1,19 @@
...
+  return new Sortable(el, {
+    animation: 150,
+    ghostClass: 'card-ghost',
+    onChoose: (e) => {
+      const handle = opts.handle ? e.item.querySelector(opts.handle) : e.item;
+      handle.classList.add('tw-cursor-grabbing');
+      opts.onChoose?.(e);
+    },
+    onUnchoose: (e) => {
+      const handle = opts?.handle ? e.item.querySelector(opts.handle) : e.item;
```
Comment (silverwind) [inline web_src/js/modules/sortable.js]:
```suggestion
      const handle = opts.handle ? e.item.querySelector(opts.handle) : e.item;
```

**Review by denyskon — APPROVED** [review]

File: web_src/js/features/repo-projects.js [hunk 5]
```diff
@@ -58,8 +58,8 @@ async function initRepoProjectSortable() {
   createSortable(mainBoard, {
     group: 'project-column',
     draggable: '.project-column',
-    animation: 150,
-    ghostClass: 'card-ghost',
+    handle: '.project-column-header',
+    filter: '[data-id="0"]',
```
Comment (silverwind) [inline web_src/js/features/repo-projects.js]:
```suggestion
```

@denyskon ah I think I have to remove this.

Comment (silverwind) [reply web_src/js/features/repo-projects.js]: Definitely, as it was removed in https://github.com/go-gitea/gitea/pull/29874.

**Review by delvh — APPROVED** [review]

**Pull request merged**